### ISSUE TITLE
DAGMC: Allow a material assignment override via dictionary on the `DAGMCUniverse` class 

### ISSUE THREAD

**Issue by pshriwise** [openmc-dev/openmc#2548]
Currently, the only way to re-establish these assignments is in Cubit or a tedious process using `PyMOAB`. I think it would be convenient to allow the `DAGMCUniverse` class to have an attribute that maps cell (CAD volume) IDs to a material object. This would make interaction with the DAGMC universe easier to work with and more consistent with how we performed these assignments (by object) elsewhere in the Python API.

If the resolution of #2547 is successful, we should also be able to reject invalid mappings on assignment of this attribute to catch mistakes early.

All of this goes for surface boundary conditions as well.

I think a partial mapping would be ok, with assignments defaulting to those found in the DAGMC file itself -- with the same requirement that at the end of the `DAGMCUniverse` initialization in C++ that the assignment mapping is complete.

**Comment by shimwell** [comment]
Agreed that does sounds handy. I recall implementing a [correspondence_dict](https://github.com/fusion-energy/openmc-dagmc-wrapper/blob/3806834e22a96f5cd58d95ff5407e83f25efebae/examples/cell_tally_example.py#L33-L43) to do something similar a while back.

I matched materials with DAGMC material tags instead of volume IDs so I will be interested to see how the volume IDs works.

Keen to see this feature
